posts/2019/11/Variable-Explorer.md

```diff
@@ -0,0 +1,73 @@
+
+It is important to mention that some of the above improvements were possible through the integration of code coming from two different projects into our own. The first of them is called [gtabview](https://github.com/TabViewer/gtabview), whose code was used for the implementation of multi-dimensional Pandas indexes. And the second one is [objbrowser](https://github.com/titusjan/objbrowser), used for viewing arbitrary Python objects.
+
+<!-- TEASER_END -->
+
+
+## New viewer for arbitrary Python objects
+
+
+For Spyder 4 we added a long-requested functionality: inspecting any kind of Python object through the Variable Explorer. For a long time Spyder has had very good support to view and edit a small subset of Python objects: Numpy arrays, Pandas dataframes and series, and collections (i.e. lists, dictionaries and tuples). Other objects were displayed in a table-like view, and inspecting any of their attributes required showing a new table,
+
+![Python Viewer](/images/spyder-variable-explorer/python-viewer.png)
+
+This made very cumbersome to use this functionality, which was the reason why arbitrary Python objects were hidden by default from the Variable Explorer view.  
+In this version we decided to integrate the excellent [objbrowser](https://github.com/titusjan/objbrowser) project, by Pepijn Kenter ([@titusjan](https://github.com/titusjan)), which provides a tree-like view of Python objects, for a much simpler and comfortable way to inspect them.
+
+![Python Viewer Metadata](/images/spyder-variable-explorer/python-viewer-metadata.png)
+
+As can be seen above, this viewer will also allow users to have access to some extra metadata about the inspected object, such as its documentation, source code and the file that holds it.  
+It is very important to note that this work was accomplished thanks to the generosity of Pepijn, who kindly changed the license of [objbrowser](https://github.com/titusjan/objbrowser) to allow us integrating it with Spyder.
+To expose this new functionality we also decided to turn off the Variable Explorer option called “Exclude unsupported types” and introduce a new one called “Exclude callables and modules” (turned on by default) to display a much larger fraction of objects after a file or cell execution.
+
+![Exclude callables modules](/images/spyder-variable-explorer/exclude-callables-modules.png)
+
+Finally, we added a context-menu action to allow using the object explorer viewer for all objects, regardless if they have a default viewer or not.
+
+![View object explorer](/images/spyder-variable-explorer/view-object-explorer.png)
+
+## Multi-index support in the dataframe viewer
+
+One of the first enhancements we did to the Variable Explorer in Spyder 4 was adding support for multi-dimensional indexes to its dataframe viewer. In Spyder 3 the support was very limited, which made inspecting those indexes a painful experience:
```

Comment:
```suggestion
One of the first features we added to the Variable Explorer in Spyder 4 was MultiIndex support in its DataFrame inspector, including for multi-level and multi-dimensional indicies. Spyder 3 had basic support, but it was very rudimentary, making inspecting such DataFrames a painful experience.
```

- Improve description
- Add some clarifications
- Fix syntax

Reply:
> had basic support **for that**, but it ...

Reply:
I edited to to read `for such`, and also switched "painful" -> "less than user-friendly" to not be too overly harsh on Spyder 3.